packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -684,27 +717,48 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     private readonly chunkMap: Map<string, string[]>;
 
     private readonly dataStores: DataStores;
-    private readonly runtimeOptions: Readonly<IContainerRuntimeOptions>;
+
+    // This is the source of truth for writing in the summary metadata blob.
+    private readonly summaryEnableGC: boolean | undefined;
+    // This is the source of truth for whether GC is enabled or not.
+    private readonly shouldRunGC: boolean;
+    /**
+     * True if generating summaries with isolated channels is
+     * explicitly disabled. This only affects how summaries are written,
+     * and is the single source of truth for this container.
+     */
+    public readonly disableIsolatedChannels: boolean;
 
     private constructor(
         private readonly context: IContainerContext,
         private readonly registry: IFluidDataStoreRegistry,
         metadata: IContainerRuntimeMetadata | undefined,
         chunks: [string, string[]][],
-        runtimeOptions: IContainerRuntimeOptions = {
-            generateSummaries: true,
-        },
+        private readonly runtimeOptions: Readonly<Required<IContainerRuntimeOptions>>,
         private readonly containerScope: IFluidObject,
         public readonly logger: ITelemetryLogger,
         private readonly requestHandler?: (request: IRequest, runtime: IContainerRuntime) => Promise<IResponse>,
         private _storage?: IDocumentStorageService,
     ) {
         super();
 
-        this.runtimeOptions = {
-            ...{ disableIsolatedChannels: true },
-            ...runtimeOptions,
-        };
+        // enableGC in metadata is introduced with v1 in the metadata blob. Force to false before that.
+        // This will override the value in runtimeOptions if it is set (true/false). So setting it in
+        // runtimeOptions will only specify what to do if it has never been set before.
+        // Note that even leaving it undefined will force it to false if no metadata blob is written.
+        const prevSummaryEnableGC = context.baseSnapshot ? gcEnabled(metadata) : undefined;
+        this.summaryEnableGC = prevSummaryEnableGC ?? this.runtimeOptions.gcOptions.summaryEnableGC;
+
+        // Can override with localStorage flag.
+        this.shouldRunGC = localStorageEnableGC() ?? (
+            // Must not be disabled permanently in summary.
+            (this.summaryEnableGC ?? true)
```

Comment:
If there is metadata blob but it does not have `enableGC` property in it, this will default to running GC right. I am wondering if that is the behavior we want.
What will happen in this scenario - A document is created before the "GC waterline" and it does not provide the `summaryEnableGC`  runtime option. But we write the metadata blob for it (maybe because of disableIsolatedChannels?) then we will end up running GC on this document right?

Reply:
Yes I thought that was correct (since we are in v1+), but now I have changed it to always write `false` for now.